nautobot/extras/views.py

```diff
@@ -695,36 +705,206 @@ def post(self, request, class_path):
             raise Http404
         job = job_class()
         grouping, module, class_name = class_path.split("/", 2)
-        form = job.as_form(request.POST, request.FILES)
+        job_form = job.as_form(request.POST, request.FILES)
+        schedule_form = forms.JobScheduleForm(request.POST)
 
         # Allow execution only if a worker process is running.
         if not get_worker_count(request):
             messages.error(request, "Unable to run job: Celery worker process not running.")
-        elif form.is_valid():
+        elif job_form.is_valid() and schedule_form.is_valid():
             # Run the job. A new JobResult is created.
-            commit = form.cleaned_data.pop("_commit")
+            commit = job_form.cleaned_data.pop("_commit")
+            schedule_type = schedule_form.cleaned_data["_schedule_type"]
+
+            if job_class.approval_required or schedule_type in choices.JobExecutionType.SCHEDULE_CHOICES:
+
+                if schedule_type in choices.JobExecutionType.SCHEDULE_CHOICES:
+                    # Schedule the job instead of running it now
+                    schedule_name = schedule_form.cleaned_data["_schedule_name"]
+                    schedule_datetime = schedule_form.cleaned_data["_schedule_start_time"]
+
+                else:
+                    # The job must be approved.
+                    # If the schedule_type is immediate, we still create the task, but mark it for approval
+                    # as a once in the future task with the due date set to the current time. This means
+                    # when approval is granted, the task is immediately due for execution.
+                    schedule_type = choices.JobExecutionType.TYPE_FUTURE
+                    schedule_datetime = datetime.now()
+                    schedule_name = f"{job.name} - {schedule_datetime}"
```

Comment:
Are we overriding a user-specified `schedule_name` from the form in this case? Or is it something that is not available in this case so we have to manually construct it?

Reply:
We have  to manually construct it, since this is for immediate executions which do not get a user-defined name.